README.md

```diff
@@ -10,7 +10,56 @@ See the [release FAQ](https://go.armory.io/release-faq).
+
+The Armory extension projects uses an OSS Service as base, currently the extension projects uses a LTS version that represents a commit in the OSS. A base artifact can be found [here](https://github.com/armory-io/orca-armory/blob/a08ea272f752ebcce01e59e790abbe0381b10d00/gradle.properties#L2) in the extension project. The base service version is maintained by Astrolabe.
+
+The base artifacts points to a commit in the OSS projects but those artifact are [jars](https://armory.jfrog.io/ui/repos/tree/General/gradle-dev-local%2Fio%2Fspinnaker) or npm packages in Artifactory. The [build ID](https://armory.jfrog.io/ui/repos/tree/Builds/gradle-dev-local%2Fio%2Fspinnaker%2Forca%2Forca-core%2F2021.07.07.21.31.05.master%2Forca-core-2021.07.07.21.31.05.master.jar) of the artifact is the commit sha from the OSS project.
+
+#### Deploy Pipeline
+
+We use Spinnaker to upgrade itself, the `Deploy RC to Staging` workflow enqueue PR checks, so we deploy & test a new service version at a time
+
+The [pipeline](https://spinnaker-staging.cloud.armory.io/#/applications/stagingcd/executions?pipeline=Deploy%20Spinnaker%20into%20SaaS%20Staging) uses a [CLI](https://github.com/armory-io/saas-tools) to extract the k8s manifests for the service changed. Then, it patch the new image tag to use and apply the manifests in the staging env. it takes the [configurations](https://github.com/armory-io/spinnaker-saas/blob/master/hosted-core-infrastructure/spinnaker-staging/kustomization.yml) from SaaS staging env.
+
+#### End-to-end pipeline tests
+
+The [pipelines](https://spinnaker-staging.cloud.armory.io/#/applications/armoryhellodeploy/executions?pipeline=Integration-Test-Runner) are the same end-to-end test that we use in preprod but we stil are migrating them. A single pipeline is responsible of trigger all the test. 
+if one of them ends with TERMINAL status, all the other pipelines stops.
+
+#### Troubleshooting
+
+When the PR check failed there could be three reasons why it's failing:
+- it fails when deploying.
+- An end-to-end pipeline failed.
+- A false alarm.
+
+The deploy & test happened in different steps in the action, if the deploy failed we can get more information in the [deploy pipeline](https://spinnaker-staging.cloud.armory.io/#/applications/stagingcd/executions?pipeline=Deploy%20Spinnaker%20into%20SaaS%20Staging).
+Also, we can take a look to the pods in the cluster to check status & logs by the service changed. You can connect to the SaaS staging using this [kubeconfig](https://github.com/armory-io/spinnaker-saas/blob/master/hosted-core-infrastructure/spinnaker-staging/kubeconfig). Under `spinnaker` namespace.
+
+```shell script
+kubectl --kubeconfig=cloud-staging-kube -n spinnaker get pods
+```
+
+If the end-to-end tests failed, we only can check which test failed searching in the end-to-end pipeline. To identify executions in Spinnaker we send as payload the serviceName, image & tag.
+You can take the logs for the service using `kubectl`
```

Comment:
Do we have logs for the staging environment in New Relic? It would be easier to point people to these instead of using kubectl.

Reply:
I agree, but I didn't see a config for that. I would like to send those logs to New Relic give a better experience to the Engineers.